src/unsafe_string.rs

```diff
@@ -54,6 +53,16 @@ trait HeaderMut<'a>: ThinMutExt<'a, Header> {
 impl<'a, T: ThinRefExt<'a, Header>> HeaderRef<'a> for T {}
 impl<'a, T: ThinMutExt<'a, Header>> HeaderMut<'a> for T {}
 
+// Constants for inline string storage
+const INLINE_STRING_MAX_LEN: usize = 7; // 7 bytes + null terminator = 8 bytes total
```

Comment:
please remove comments regarding null termination. it's not null terminated. the 8th byte is not 0. At minimum, it's InlineString, which is 7, OR'd with the length information in the remaining 5 bits.

(And it cannot be 0, because we require that all-0 be a niche bit representation that can be filled by a None Option, exactly as NonNull does.)